.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,105 @@
+    Build:
+        strategy:
+            matrix:
+                options:
+                    - os: ubuntu-latest
+                      framework: netcoreapp3.1
+                      runtime: -x64
+                      codecov: false
+                    - os: windows-latest
+                      framework: netcoreapp3.1
+                      runtime: -x64
+                      codecov: true
+                    - os: windows-latest
+                      framework: netcoreapp2.1
+                      runtime: -x64
+                      codecov: false
+
+        runs-on: ${{matrix.options.os}}
+
+        steps:
+            - uses: actions/checkout@v2
+
+            - name: Install NuGet
+              uses: NuGet/setup-nuget@v1
+
+            - name: Setup Git
+              shell: bash
+              run: |
+                  git config --global core.autocrlf false
+                  git config --global core.longpaths true
+                  git fetch --prune --unshallow
+                  git submodule -q update --init --recursive
+
+            - name: Fetch Tags for GitVersion
+              run: |
+                  git fetch --tags
+
+            - name: Fetch master for GitVersion
+              if: github.ref != 'refs/heads/master'
+              run: git branch --create-reflog master origin/master
+
+            - name: Install GitVersion
+              uses: gittools/actions/setup-gitversion@v0.3
+              with:
+                  versionSpec: "5.1.x"
+
+            - name: Use GitVersion
+              id: gitversion # step id used as reference for output values
+              uses: gittools/actions/execute-gitversion@v0.3
+
+            - name: Setup DotNet SDK
+              uses: actions/setup-dotnet@v1
+              with:
+                  dotnet-version: "3.1.101"
+
+            - name: Setup Azurite
+              shell: bash
+              run: sudo npm install -g azurite
+
+            - name: Use Azurite
+              shell: bash
+              run: sudo azurite --loose &
+
+            - name: Build
+              shell: pwsh
+              run: ./ci-build.ps1 "${{steps.gitversion.outputs.nuGetVersion}}" "${{matrix.options.framework}}"
+
+            - name: Test
+              shell: pwsh
+              run: ./ci-test.ps1 "${{matrix.options.os}}" "${{matrix.options.framework}}" "${{matrix.options.runtime}}" "${{matrix.options.codecov}}"
+              env:
+                  CI: True
+                  XUNIT_PATH: .\tests\ImageSharp.Web.Tests # Required for xunit
+
+            - name: Update Codecov
+              uses: iansu/codecov-action-node@v1.0.0
+              if: matrix.options.codecov == true
+              with:
+                  token: ${{secrets.CODECOV_TOKEN}}
+                  file: "coverage.${{matrix.options.framework}}.xml"
+                  flags: unittests
+
+            - name: Pack # We can use this filter as we know it happens only once and takes the most time to complete.
```

Comment:
we should be using the split build/publish workflow here not skip and hope it takes the longest

Reply:
Yeah, good spot. I'll update that once I get it working.